modal: Make the modal content scrollable instead of the whole modal.

ganpa3 opened the pull request:
See https://chat.zulip.org/#narrow/stream/6-frontend/topic/Modal.20bug/near/1437163.

### Before

### Light mode

![image](https://user-images.githubusercontent.com/58626718/192215424-f8306fa2-2016-4e5c-8bcd-51fbf30cb2c9.png)

![image](https://user-images.githubusercontent.com/58626718/192214800-f7679767-8c48-4ba0-93c6-01c2e972e07b.png)

### Dark mode

![image](https://user-images.githubusercontent.com/58626718/191990750-ec92fc13-be9f-433c-b9e7-6236567cf2f7.png)

![image](https://user-images.githubusercontent.com/58626718/191990550-ee9bf465-7922-449a-9892-7c43607d2571.png)

### After

### Light mode
![image](https://user-images.githubusercontent.com/58626718/192215075-dba824dd-98fb-4828-9c36-3fc66c796752.png)

![image](https://user-images.githubusercontent.com/58626718/192214800-f7679767-8c48-4ba0-93c6-01c2e972e07b.png)

### Dark mode

![image](https://user-images.githubusercontent.com/58626718/191990451-aa6a4922-3608-414f-918f-2d8d4f64db40.png)

![image](https://user-images.githubusercontent.com/58626718/191990550-ee9bf465-7922-449a-9892-7c43607d2571.png)

alya commented:
The screenshots above look good to me! Just to check, can you also post screenshots for light theme? Thanks!

ganpa3 commented:
> The screenshots above look good to me! Just to check, can you also post screenshots for light theme? Thanks!

Done.

alya commented:
Cool, looks good to me when doing a bit of manual testing (and in the screenshots above)!

@timabbott you may want to prioritize reviewing this one, as it blocks getting some other changes deployed.

timabbott commented on static/js/dialog_widget.js:
```diff
@@ -157,7 +158,24 @@ export function launch(conf) {
+                } else {
+                    $modal_footer.css("border-top", "1px solid hsl(0, 0%, 87%)");
+                }
+            }
```
There's a few things we need to change here:
* We should set a `scrollable_modal` CSS class, it would be better to have `border-top` styling for the two themes in CSS files, rather than doing inline `.css` here. Among other things, doing so will live-update properly, and also just have all the CSS remain in CSS files.
* Some modals, like Read Receipts, can change in height after being initially loaded (when a network request returns)
* All modals can have the browser window be resized while they're open; and this logic likely won't handle that.

A pure CSS solution would be a lot better; and I think it might be possible; there should be something like a `:visible` selector we can do on a simplebar generated container for whether a scrollbar is visible or not.

ganpa3 replied:
Not sure what a pure CSS solution would be. I went with using a `MutationObserver` on the vertical scrollbar. Updated the PR.

timabbott commented:
@sahil839 can you help review this? I remember we had to revert the original version of this because while it improved the situation for very tall modals, we had some issues with the look on smaller ones, but I don't remember the context very well and would like to avoid regressions.

sahil839 commented:
I see that simplebar adds `simplebar-scrollable-y` class to the div containing modal content, so we can use `+` operator to have a only CSS solution to add border.

ganpa3 commented:
@sahil839 Thanks for the suggestion. Updated the PR.

@timabbott Ready for review.

timabbott commented:
This looks very reasonable in terms of the code; @alya it's probably worth your doing a manual QA pass on this just because it's something where we had issues with after merging a previous version in the past.

timabbott commented:
Test-deploying this on chat.zulip.org.

alya commented:
I'm seeing a regression with the "Move messages" modal in really short windows. Visually, one can get weird effects:

<img width="713" alt="Screen Shot 2023-02-24 at 2 06 15 PM" src="https://user-images.githubusercontent.com/2090066/221302547-28ac7264-264d-43b9-a3dc-8686c9f91eac.png">

And then if the window is really short, there is no way to view the buttons at all:

<img width="571" alt="Screen Shot 2023-02-24 at 2 00 27 PM" src="https://user-images.githubusercontent.com/2090066/221302611-f359339c-71fb-46e5-804f-0344b474eed9.png">

On CZO, if I make the window really short, I get a scroll bar on the modal, which is better.

ganpa3 commented:
> I'm seeing a regression with the "Move messages" modal in really short windows. Visually, one can get weird effects:
> 
> <img alt="Screen Shot 2023-02-24 at 2 06 15 PM" width="713" src="https://user-images.githubusercontent.com/2090066/221302547-28ac7264-264d-43b9-a3dc-8686c9f91eac.png">
> 
> And then if the window is really short, there is no way to view the buttons at all:
> 
> <img alt="Screen Shot 2023-02-24 at 2 00 27 PM" width="571" src="https://user-images.githubusercontent.com/2090066/221302611-f359339c-71fb-46e5-804f-0344b474eed9.png">
> 
> On CZO, if I make the window really short, I get a scroll bar on the modal, which is better.

This occurs because the `overflow` property is [set to be `visible`](https://github.com/zulip/zulip/blob/a2ea5e859875e033dd93c88efd0c6e0f57d07735/web/styles/popovers.css#L796-L798), which is done to [prevent the topic dropdown list from being cut off the modal](https://github.com/zulip/zulip/pull/17216#discussion_r610015517).

Removed that property since it isn't needed now and updated the PR. Ready for review.

Screenshot

![image](https://user-images.githubusercontent.com/58626718/221361793-7233129d-9772-476e-85fb-853057b3c802.png)

timabbott commented:
This works well in my testing, merged, thanks so much for getting to a clean fix here @ganpa3!
